### PR TITLE
Kara's updates

### PR DESCRIPTION
**Pull request by karafelix**
I started a new branch, starting from your current master branch, and redid the changes I made--adding the integration tests, modifying delete_stemcell to not delete it, and using the environment variable to determine the datacenter.
I also fixed up a few more things that were making it fail tests.

### CODE REVIEW

File: integration/delete_stemcell/delete_stemcell_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	testhelperscpi "github.com/maximilien/bosh-softlayer-cpi/test_helpers"
+	slclient "github.com/maximilien/softlayer-go/client"
+	datatypes "github.com/maximilien/softlayer-go/data_types"
+	softlayer "github.com/maximilien/softlayer-go/softlayer"
+	// testhelpers "github.com/maximilien/softlayer-go/test_helpers"
```
Comment (maximilien) [inline integration/delete_stemcell/delete_stemcell_test.go]: Get rid of this comment

**Comment by maximilien** [comment]
Minor. Please address comment and will merge ASAP.

**Comment by karafelix** [comment]
Yep, that was just an artifact of when I was first trying to get it to work. Done.
This does raise the question to me though, would it be better to be using the delete_stemcell test fixture json file instead of generating the json in the code like this?

**Comment by maximilien** [comment]
Either is OK but I would follow the rest of code by having a fixture. One other think I see is in line 95 and a few other places you are waiting for a minute for transaction to be done. Id rather see small wait in a bounded loop checking instead. I think the test helpers is doing that now. Please see if you can change to that and update PR. Thanks.

**Comment by karafelix** [comment]
I believe it should now be properly waiting for the transaction to complete. I don't remember adding the sleep statement, so it probably came from somewhere else in code I was modifying, so I looked to see if it was present in other places. All I found was in create_stemcell and delete_stemcell, both of which I modified.

**Comment by maximilien** [comment]
Ok cool. Cheers :beers:

**Pull request merged**